tsdb/engine/tsm1/file_store.go

```diff
@@ -845,6 +845,10 @@ func ParseTSMFileName(name string) (int, int, error) {
 	}
 
 	generation, err := strconv.ParseUint(id[:idx], 10, 32)
+	if err != nil {
+		return 0, 0, err
```

Comment:
In the unlikely event we hit this error case, the raw error from parsing a uint will be confusing. Please wrap this error and the next one, like in https://github.com/influxdata/influxdb/pull/7747/files#diff-49448274394de97159e082c06d9532a1R846.